spring-cloud-open-service-broker-core/src/main/java/org/springframework/cloud/servicebroker/controller/ServiceInstanceBindingController.java

```diff
@@ -107,7 +108,15 @@ public ServiceInstanceBindingController(CatalogService catalogService,
 						.doOnSuccess(response -> LOG.debug("Creating a service instance binding succeeded: serviceInstanceId={}, bindingId={}, response={}",
 								serviceInstanceId, bindingId, response)))
 				.map(response -> new ResponseEntity<>(response, getCreateResponseCode(response)))
-				.switchIfEmpty(Mono.just(new ResponseEntity<>(HttpStatus.CREATED)));
+				.switchIfEmpty(Mono.just(new ResponseEntity<>(HttpStatus.CREATED)))
+				.onErrorResume(e -> {
+					if (e instanceof ServiceDefinitionDoesNotExistException ||
+							e instanceof ServiceDefinitionPlanDoesNotExistException) {
+						return Mono.just(new ResponseEntity<>(HttpStatus.BAD_REQUEST));
```

Comment:
Apparently I'm stuck, here I need to return an error message.

Reply:
Apparently I will have to handle it globally. Although the method of processing locally I liked more.